src/jennifer/model/column_mapping.cr

```diff
@@ -0,0 +1,46 @@
+module Jennifer::Model
+  module ColumnMapping
+    #
+    # This macro defines a constant `COLUMN_MAPPINGS` that captures
+    # all mappings from field names to column names that were defined in the
+    # given property specification.
+    # Those mappings may be specified by specifying the mapping attribute
+    # `column` in combination with the targetted column name.
+    # E.g. a field mapping like `a: { type: Int64, column: :b}`
+    # results in a field `a` that is mapped to the column `b` in the database.
+    # The mapping constant is accessible in other macros and resolves field
+    # names to column names. In order to generalize access, it also stores
+    # fields that do not have a custom mapping - in this case the map
+    # returns the same name as the field (e.g. for above example
+    # `COLUMN_MAPPINGS["a"] == "b"`, but for another - unmapped column - `c`
+    # it holds that `COLUMN_MAPPINGS["c"] == "c"`).
+    #
+    # @param properties All property definitions of the mapped type
+    #
+    macro _register_column_mappings(properties)

```

Comment:
Would this also work with STI and validations? That was my main concern when designing this module / macro ... The `Uniqueness` validation as an example requires the mapping to look up the actual name of the column (see [here](https://github.com/skloibi/jennifer.cr/blob/feature/property_aliases/src/jennifer/validations/macros.cr#L90)), otherwise the developer has to use the real column names when specifying uniqueness validations.
Similarly for STI mappings: To my knowledge, the `properties` maps of class hierarchies are not merged, therefore the subclass cannot access the real column names of properties defined in the superclass. Those are required for the `update_columns` method in `STIMapping` ([here](https://github.com/skloibi/jennifer.cr/blob/feature/property_aliases/src/jennifer/model/sti_mapping.cr#L176)). Or did I misunderstands something there?

Reply:
1. `Uniqueness` validator can use `Model._field_alias` instead of `_field_name` in the query which returns original column name
2. STI merges `COLUMNS_METADATA` at the very end of macro. Also `#update_columns` calls super with all missing at the current "level" fields

Reply:
I've finished my work with mapping refactoring - I can wait for your PR and fix all conflicts or (if you prefer) merge it before this

Reply:
Nice, thanks for the clarification. I will try to implement those changes this coming week!
Regarding the merging order, I don't really mind. As this change is primarily a replacement of `key` accesses, I can surely wait until your refactoring is included. That way you also don't have to wade through my test cases. But if you prefer it the other way, I am also fine with that!